tables/core/group.py

```diff
@@ -187,17 +220,117 @@ def __contains__(self, item):
+        if not hasattr(where, 'create_array'):
+            where = self._get_or_create_path(where, createparents)
         return where.create_array(*args, **kwargs)
 
-    def create_group(self, where, *args, **kwargs):
+    def create_group(self, where, *args, createparents=False, **kwargs):
+        if not hasattr(where, 'create_group'):
+            where = self._get_or_create_path(where, createparents)
         return where.create_group(*args, **kwargs)
 
-    def create_table(self, where, name, desc, *args, **kwargs):
+    def create_table(self, where, name, desc, *args, createparents=False, **kwargs):
+        if not hasattr(where, 'create_table'):
+            where = self._get_or_create_path(where, createparents)
         return where.create_table(name, desc, *args, **kwargs)
 
-    def get_node(self, where):
-        return self.root[where]
+    def get_node(self, where, name=None, classname=None):
+        """Get the node under where with the given name.
+
+        Parameters
+        ----------
+        where : str or Node
+            This can be a path string leading to a node or a Node instance (see
+            :ref:`NodeClassDescr`). If no name is specified, that node is
+            returned.
+
+            .. note::
+
+                If where is a Node instance from a different file than the one
+                on which this function is called, the returned node will also
+                be from that other file.
+
+        name : str, optional
+            If a name is specified, this must be a string with the name of
+            a node under where.  In this case the where argument can only
+            lead to a Group (see :ref:`GroupClassDescr`) instance (else a
+            TypeError is raised). The node called name under the group
+            where is returned.
+        classname : str, optional
+            If the classname argument is specified, it must be the name of
+            a class derived from Node (e.g. Table). If the node is found but it
+            is not an instance of that class, a NoSuchNodeError is also raised.
+
+        If the node to be returned does not exist, a NoSuchNodeError is
+        raised. Please note that hidden nodes are also considered.
+
+        """
+
+        self._check_open()
+
+        if isinstance(where, Group):
+            where._g_check_open()
+
+            node = where[name]
+        elif isinstance(where, (six.string_types, np.str_)):
+            if not where.startswith('/'):
+                raise NameError("``where`` must start with a slash ('/')")
+
+            basepath = where
+            nodepath = join_path(basepath, name or '') or '/'
+            node = self.root[nodepath]
+        else:
+            raise TypeError(
+                "``where`` must be a string or a node: %r" % (where,))
+
+        # Finally, check whether the desired node is an instance
+        # of the expected class.
+        if classname:
```

Comment:
might be worth checking `classname is not None` so other 'falsy' values (which should probably raise) do not safely pass.